src/locale/localed-util.c

```diff
@@ -699,6 +697,38 @@ int vconsole_convert_to_x11(const VCContext *vc, X11Context *ret) {
 
                 return x11_context_copy(ret, &xc);
         }
+
+        /* No custom mapping has been found, see if it's the keymap is a converted one as in such case deduce
+         * the conversion to the corresponding x11 layout is easy. */
+        _cleanup_free_ char *xlayout = NULL, *converted = NULL;
+        char *xvariant = NULL;
+
+        xlayout = strdup(vc->keymap);
+        if (!xlayout)
+                return -ENOMEM;
+        xvariant = strchr(xlayout, '-');
+        if (xvariant)
+                *xvariant++ = '\0';
```

Comment:
this is unnecessarily hard to read. at least add `()` to make sure what is applied first. or better: just do two lines. no need to by unnecessarily cryptic